src/main/java/com/google/api/codegen/transformer/ModelToJavaSurfaceTransformer.java

```diff
@@ -0,0 +1,255 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.ApiConfig;
+import com.google.api.codegen.InterfaceView;
+import com.google.api.codegen.LanguageUtil;
+import com.google.api.codegen.MethodConfig;
+import com.google.api.codegen.PageStreamingConfig;
+import com.google.api.codegen.ServiceConfig;
+import com.google.api.codegen.gapic.GapicCodePathMapper;
+import com.google.api.codegen.viewmodel.ApiCallableView;
+import com.google.api.codegen.viewmodel.ApiMethodView;
+import com.google.api.codegen.viewmodel.BundlingApiCallableView;
+import com.google.api.codegen.viewmodel.PagedApiCallableView;
+import com.google.api.codegen.viewmodel.SimpleApiCallableView;
+import com.google.api.codegen.viewmodel.StaticXApiView;
+import com.google.api.codegen.viewmodel.StaticXSettingsView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.Model;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ModelToJavaSurfaceTransformer implements ModelToViewTransformer {
+  private ApiConfig cachedApiConfig;
+  private GapicCodePathMapper pathMapper;
+  private PathTemplateTransformer pathTemplateTransformer;
+  private ApiMethodTransformer apiMethodTransformer;
+
+  private static final String XAPI_TEMPLATE_FILENAME = "java/xapi.snip";
+  private static final String XSETTINGS_TEMPLATE_FILENAME = "java/xsettings.snip";
+
+  public ModelToJavaSurfaceTransformer(ApiConfig apiConfig, GapicCodePathMapper pathMapper) {
+    this.cachedApiConfig = apiConfig;
+    this.pathMapper = pathMapper;
+    this.pathTemplateTransformer = new PathTemplateTransformer();
+    this.apiMethodTransformer = new ApiMethodTransformer();
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(XAPI_TEMPLATE_FILENAME, XSETTINGS_TEMPLATE_FILENAME);
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model) {
+    List<ViewModel> surfaceDocs = new ArrayList<>();
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      surfaceDocs.addAll(transform(service));
+    }
+    return surfaceDocs;
+  }
+
+  public List<ViewModel> transform(Interface service) {
+    SurfaceTransformerContext context =
+        SurfaceTransformerContext.create(
+            service, cachedApiConfig, new ModelToJavaTypeTable(), new JavaSurfaceNamer());
```

Comment:
It's passed all over the place to the other transformers:

* ApiMethodTransformer
* InitCodeTransformer
* PageStreamingTransformer
* PathTemplateTransformer

It's not meant to be a transformer context for all transformers; e.g. Shin's test transformers would use their own context. It is as specific as it needs to be.

I don't think we should be caching api config, since we want to support batch code generation over multiple APIs later. Also we shouldn't make the surface namer into a singleton either, if we want to support batch code generation over multiple languages.